sql/core/src/test/scala/org/apache/spark/sql/expressions/ExpressionInfoSuite.scala

```diff
@@ -191,4 +191,85 @@ class ExpressionInfoSuite extends SparkFunSuite with SharedSparkSession {
+      "org.apache.spark.sql.catalyst.expressions.GreaterThanOrEqual",
+      "org.apache.spark.sql.catalyst.expressions.Greatest",
+      "org.apache.spark.sql.catalyst.expressions.If",
+      "org.apache.spark.sql.catalyst.expressions.In",
+      "org.apache.spark.sql.catalyst.expressions.Inline",
+      "org.apache.spark.sql.catalyst.expressions.InputFileBlockLength",
+      "org.apache.spark.sql.catalyst.expressions.InputFileBlockStart",
+      "org.apache.spark.sql.catalyst.expressions.InputFileName",
+      "org.apache.spark.sql.catalyst.expressions.JsonTuple",
+      "org.apache.spark.sql.catalyst.expressions.Least",
+      "org.apache.spark.sql.catalyst.expressions.LessThan",
+      "org.apache.spark.sql.catalyst.expressions.LessThanOrEqual",
+      "org.apache.spark.sql.catalyst.expressions.MapKeys",
+      "org.apache.spark.sql.catalyst.expressions.MapValues",
+      "org.apache.spark.sql.catalyst.expressions.Md5",
+      "org.apache.spark.sql.catalyst.expressions.MonotonicallyIncreasingID",
+      "org.apache.spark.sql.catalyst.expressions.Multiply",
+      "org.apache.spark.sql.catalyst.expressions.Murmur3Hash",
+      "org.apache.spark.sql.catalyst.expressions.Not",
+      "org.apache.spark.sql.catalyst.expressions.Or",
+      "org.apache.spark.sql.catalyst.expressions.Overlay",
+      "org.apache.spark.sql.catalyst.expressions.Pmod",
+      "org.apache.spark.sql.catalyst.expressions.PosExplode",
+      "org.apache.spark.sql.catalyst.expressions.Remainder",
+      "org.apache.spark.sql.catalyst.expressions.Sha1",
+      "org.apache.spark.sql.catalyst.expressions.Sha2",
+      "org.apache.spark.sql.catalyst.expressions.Size",
+      "org.apache.spark.sql.catalyst.expressions.SortArray",
+      "org.apache.spark.sql.catalyst.expressions.SparkPartitionID",
+      "org.apache.spark.sql.catalyst.expressions.Stack",
+      "org.apache.spark.sql.catalyst.expressions.Subtract",
+      "org.apache.spark.sql.catalyst.expressions.TimeWindow",
+      "org.apache.spark.sql.catalyst.expressions.UnaryMinus",
+      "org.apache.spark.sql.catalyst.expressions.UnaryPositive",
+      "org.apache.spark.sql.catalyst.expressions.Uuid",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathBoolean",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathDouble",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathFloat",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathInt",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathList",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathLong",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathShort",
+      "org.apache.spark.sql.catalyst.expressions.xml.XPathString"
+    )
+    val sinceMissingExprs = spark.sessionState.functionRegistry.listFunction()
+      .map(spark.sessionState.catalog.lookupFunctionInfo)
+      .filterNot(e => ignoreSet.contains(e.getClassName))
+      .filter(funcInfo => !funcInfo.getSince.matches("[0-9]+\\.[0-9]+\\.[0-9]+"))
```

Comment:
@maropu, I think the asserts in the constructor of `ExpressionInfo` would cover what this PR aims. About ensuring to have the `since` for new expressions, I think it's less likely that we miss adding `since` during the review of other PRs just given my experience.

Reply:
okay, I will close this. Thanks for the check, @HyukjinKwon